nemo_skills/code_execution/local_sandbox/start_local_sandbox.sh

```diff
@@ -17,7 +17,24 @@
 # NOTE: needs to run from the root of the repo!
 
 SANDBOX_NAME=${1:-'local-sandbox'}
+DEPLOYMENT_MODE=${DEPLOYMENT_MODE:-'single'}
 
-docker build --tag=${SANDBOX_NAME} --build-arg="UWSGI_PROCESSES=$((nproc --all * 10))" --build-arg="UWSGI_CHEAPER=nproc --all" -f dockerfiles/Dockerfile.sandbox .
+docker build --tag=${SANDBOX_NAME} --build-arg="UWSGI_PROCESSES=$((`nproc --all` * 10))" --build-arg="UWSGI_CHEAPER=nproc --all" -f dockerfiles/Dockerfile.sandbox .
 
-docker run --network=host --rm --memory=${NEMO_SKILLS_SANDBOX_MEM_LIMIT:-"16g"} --restart unless-stopped --name=local-sandbox ${SANDBOX_NAME}
+echo "Starting sandbox in $DEPLOYMENT_MODE mode..."
+
+DEPLOYMENT_ARGS=""
+
+if [ "$DEPLOYMENT_MODE" = "multi-worker" ]; then
```

Comment:
Do we need two different modes? Is there a downside in using multi-worker always?

Reply:
So at the moment, if given enough resources, there is very little performance difference between the two modes. But if there code being executed consumes a lot of memory, it will tend to lead to a timeout in the multi-worker setting (which is time consuming) rather than failing quickly like the previous single deployment method. I'll see if I can tweak it to avoid this behavior, in which case, I would feel more comfortable moving to multi-worker completely.

Reply:
The other difference would be the job scheduling. 'single' deployment mode has a slight edge because it can assign requests to the first available worker, effectively balancing workloads across workers. Whereas the 'multi-worker' version with session-affinity will assign a job to a replica without regard the current workload of that replica, even when there is no session-id. This could lead to some inefficiency in scheduling.